tempto-core/src/main/java/com/teradata/tempto/hadoop/FileSystemClient.java

```diff
@@ -45,6 +45,8 @@ InputStream getInputStream()
 
     void createDirectory(String path);
 
+    void deleteDirectory(String path);
```

Comment:
what is a difference between these two methods? If first is able to delete directories and second only files, then second should fail when getting a directory, so you cannot implement the first one with the second.

As I understand previous semantic was to delete always (files and directories), was there any issue with such approach? Can we leave it as it was?

Reply:
At first I thought that this was not possible because of the way directories are represented in S3 and Azure but now I think this is possible.  I will investigate.